src/qmc_states.jl

```diff
@@ -124,7 +124,8 @@ num_spectral_states(::ReplicaState{<:Any, S}) where {S} = S
 """
     num_overlaps(state_or_DataFrame)
 
-Return the number of overlaps between replicas.
+Return the number of overlaps between replicas. Only counts overlaps between replicas of
+the same spectral state, even if `AllOverlaps` is used with `mixed_spectral_overlaps=true`.
```

Comment:
```suggestion
Return the number of overlaps between replicas that are being reported. Only counts 
overlaps between replicas of the same spectral state, even if `AllOverlaps` is used with 
`mixed_spectral_overlaps=true`.

The return value is `0` if no overlaps are being reported, and `N*(N-1)÷2` otherwise, where `N` is the value returned by [`num_replicas`](@ref).

See also [`ProjectorMonteCarloProblem`](@ref), [`AllOverlaps`](@ref), [`num_spectral_states`](@ref).
```

Can we also please make `num_overlaps` work on instances of `ProjectorMonteCarloProblem` to make it consistent with `num_replicas`?